modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java

```diff
@@ -82,7 +85,8 @@ protected AionRepositoryImpl(IRepositoryConfig repoConfig) {
         private static final AionRepositoryImpl inst =
                 new AionRepositoryImpl(
                         new RepositoryConfig(
-                                new File(config.getBasePath(), config.getDb().getPath())
+                                // Allow path to be set by --datadir or config file
+                                new File(config.getBasePath(), dbPath)
```

Comment:
I don't like that the path is set while bypassing the RepositoryConfig. I feel like exposing this functionality is very risky. Let's clarify the specifications to figure out if this is really necessary. @AionJayT @qoire @yulongaion

Reply:
This will be changed so that the config.xml is overwritten when user enters "--datadir [directory]" instead of accessing the database implementation